src/picongpu/include/plugins/adios/restart/LoadSpecies.hpp

```diff
@@ -126,20 +124,15 @@ struct LoadSpecies
         ADIOS_CMD(adios_perform_reads( params->fp, 1 ));
         adios_selection_delete(piSel);
 
-        /* search my entry (using my scalar position) in particlesInfo */
+        /* Run a prefix sum over the numParticles[0] element in particlesInfo
+         * to retreive the offset of particles before gc.getScalarPosition() */
         uint64_t particleOffset = 0;
-        uint64_t myScalarPos = gc.getScalarPosition();
+        totalNumParticles = particlesInfo[0];
 
-        for (size_t i = 0; i < gc.getGlobalSize(); ++i)
-        {
-            if (particlesInfo[i][1] == myScalarPos)
-            {
-                totalNumParticles = particlesInfo[i][0];
-                break;
-            }
-
-            particleOffset += particlesInfo[i][0];
-        }
+        MPI_CHECK(MPI_Exscan( &totalNumParticles, &particleOffset, 1, MPI_UINT64_T, MPI_SUM, params->adiosComm ));
+        /* Rank 0 needs special treatmend in `MPI_Exscan` */
+        if( gc.getScalarPosition() == 0 )
```

Comment:
This is for sanity. I am not sure if `undefined` means "guaranteed to be not overwritten" in [this manual](https://www.open-mpi.org/doc/v1.8/man3/MPI_Exscan.3.php#toc7)